src/store/middleware/camerasMiddleware.js

```diff
@@ -74,11 +87,26 @@ const camerasMiddleware = store => next => action => {
 
     case messageReceivedFromRover.type: {
       const { message } = action.payload;
-      if (message.type === "cameraStreamReport")
+      if (message.type === "cameraStreamReport") {
         store.dispatch(cameraStreamDataReportReceived({
           cameraName: message.camera,
           frameData: message.data
         }));
+      } else if (message.type === "cameraFrameReport") {
+        if (message.data !== "") {
```

Comment:
Should be combined into `else if (message.type === ... && message.data === ...)`